### PR TITLE
Replace find with forEach

### PR DESCRIPTION
**Pull request by ciss1995**

### CODE REVIEW

File: src/util.js [hunk 1]
```diff
@@ -3,8 +3,11 @@ export function contains(arrayLike, stringOrRegex) {
 		return arrayLike.indexOf(stringOrRegex) > -1;
 	}
 
-	const match = arrayLike.find(function(item) {
-		return item.match(stringOrRegex);
+	var match;
+	arrayLike.forEach(function(item) {
+		if (item.match(stringOrRegex)) {
+			match = item.match(stringOrRegex);
+		}
```
Comment (dlockhart) [inline src/util.js]: I think previously `find()` would have returned the first match, whereas now `match` will be set to the last match.

Comment (ciss1995) [reply src/util.js]: Agreed! Good catch

**Comment by ctwomey1** [comment]
Any reason we didn't want to just polyfill .find in the relevant places?

File: src/util.js (hunk 1 above)
Comment (dlockhart) [reply src/util.js]: As strange as it seems, this only returns from the inner `function` -- the `forEach` will still iterate over every entry. There's no easy way to short-circuit a `forEach` (without throwing an exception). I'd recommend a normal `for`-loop:

```
let match;
for (let i=0; i<arrayLike.length; i++) {
    match = arrayLike[i].match(stringOrRegex)) {
    if (match) break;
}
```

**Comment by dlockhart** [comment]
I wasn't a huge fan of polyfilling here because:
1) Typically it's up to consumers to polyfill, otherwise the polyfill could be included many times. In this case, consumers weren't even aware that they needed to polyfill... how do we find all the places using this and add it in?
2) Not using `array.find` only adds a couple lines... i.e. it doesn't really save much so seems like it isn't even worth the trouble.

**Comment by ctwomey1** [comment]
Yeah I agree polyfilling here doesn't work. Was mostly just curious

I had thought we just needed to polyfill in BSI since this was directly needed for a my courses issue, but you're right it could be used in many places around brightspace or any potential external consumers.

One day ie11 won't exist and we no longer need to jump through hoops

File: src/util.js (hunk 1 above)
Comment (dougmoscrop) [reply src/util.js]: Could also use `reduce` if you don't care about the early out

```js
const match = arrayLike.reduce(function(memo, item) {
  return memo || item.match(stringOrRegex);
}, undefined);
```

File: src/util.js [hunk 2]
```diff
@@ -3,8 +3,11 @@ export function contains(arrayLike, stringOrRegex) {
 		return arrayLike.indexOf(stringOrRegex) > -1;
 	}
 
-	const match = arrayLike.find(function(item) {
-		return item.match(stringOrRegex);
+	var match;
+	arrayLike.forEach(function(item) {
+		if (item.match(stringOrRegex)) {
+			return match = item.match(stringOrRegex);
```
Comment (ryantmer) [inline src/util.js]: Functionally the same, but splitting this into two statements (the assignment and the `return`) might be clearer - might just be me, but returning an assignment tripped me up for a second! (Just not something I commonly see is all.)

**Pull request merged**